fiftyone/operators/cache/utils.py

```diff
@@ -0,0 +1,204 @@
+"""
+Execution cache utils.
+
+| Copyright 2017-2025, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
+"""
+from cachetools.keys import hashkey
+import numpy as np
+import json
+import hashlib
+
+import eta.core.serial as etas
+
+import fiftyone as fo
+from fiftyone.operators.store import ExecutionStore
+
+
+def get_function_id(func):
```

Comment:
Need to clean this up and:

 - decide which are public vs private
 - add doc strings to public methods